internal/federation/handle.go

```diff
@@ -194,3 +194,39 @@ func HandleKeyRequests() func(*Server) {
 		keymux.Handle("/server/{keyID}", keyFn).Methods("GET")
 	}
 }
+
+// HandleMediaRequests is an option which will process /_matrix/media/v1/* requests using predefined media IDs as ways to
+// get particular kinds of media. The origin of the media must match the server name. Uploads are not handled. The
+// predefined media IDs are:
+//   * PlainTextFile   => "Hello from the other side"
+func HandleMediaRequests() func(*Server) {
```

Comment:
Rather than have fixed responses like that, it'd be nice if this function accepted a map as a function arg which did this e.g:
```go
func HandleMediaRequests(files map[string][]byte) func(*Server) {
```
usage:
```go
HandleMediaRequests(map[string][]byte{
    "PlainTextFile": []byte("Hello from the other side"),
})
```

Reply:
This does imply we just use `application/octet-stream` everywhere for the content type.

Reply:
We'd need the content type to be set to specific things alongside other headers for other tests, so this is likely going to have to be a `map[string]Response` or something (maybe just a function to avoid pigeon holing?)

Reply:
Yeah a new function SGTM